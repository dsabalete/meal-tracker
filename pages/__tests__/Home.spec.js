import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import HomeView from '@/pages/index.vue'

const build = (props) => {
  const wrapper = shallowMount(HomeView, {
    stubs: {
      NuxtLink: RouterLinkStub,
    },
    propsData: { ...props },
  })
  return { wrapper }
}

describe('Home View', () => {
  test('should be rendered', () => {
    const { wrapper } = build()
    expect(wrapper.exists()).toBeTruthy()
  })

  test('should match snapshot', () => {
    const { wrapper } = build()
    expect(wrapper.element).toMatchSnapshot()
  })

  test('', () => {})
})
