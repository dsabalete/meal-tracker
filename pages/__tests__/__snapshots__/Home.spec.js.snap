// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home View should match snapshot 1`] = `
<div
  class="flex flex-col flex-grow space-y-12 text-blue-400 max-h-full"
>
  <colorswitch-stub />
   
  <div
    class="logo mx-auto"
  >
    <mealtrackericon-stub
      class="w-96 h-96"
    />
  </div>
   
  <div
    class="mx-auto"
  >
    <h1
      class="font-sans font-bold text-4xl"
    >
      MealTracker
    </h1>
  </div>
   
  <div
    class="p-4 w-2/3 mx-auto shadow-lg rounded-lg bg-white dark:bg-gray-700 text-center"
  >
    <a
      class="text-blue-400 dark:text-blue-100 font-bold text-2xl"
    >
      
      Login
    
    </a>
  </div>
   
  <div
    class="p-4 w-2/3 mx-auto shadow-lg rounded-lg bg-blue-400 dark:bg-blue-700 text-center"
  >
    <a
      class="text-white dark:text-blue-100 font-bold text-2xl"
    >
      
      Create account
    
    </a>
  </div>
</div>
`;
